core/src/main/resources/org/apache/spark/ui/static/stagepage.js

```diff
@@ -473,27 +492,6 @@ $(document).ready(function () {
             var accumulatorTable = responseBody.accumulatorUpdates.filter(accumUpdate =>
                 !(accumUpdate.name).toString().includes("internal."));
 
-            // rendering the UI page
-            var data = {"executors": response};
```

Comment:
Maybe a dumb question, but where did this go then - how is the `data` loaded and displayed?

Reply:
Before Fix- In existing code

`var data = {"executors": response};
            $.get(createTemplateURI(appId, "stagespage"), function(template) {
                tasksSummary.append(Mustache.render($(template).filter("#stages-summary-template").html(), data));`

In code response object comes from this api call ("$.getJSON(endPoint + "/" + stageAttemptId, function(response, status, jqXHR)") and this API call is invoked before the call made to get the html page of stage and then adding that response in the html page.

After Fix 

The stage page html is first loaded and all the api calls are inside the `$.get(createTemplateURI(appId, "stagespage"), function(template) {` , So we dont need to add the response here in this scenario.